### PR TITLE
[DC-1544] Do not start processing a submission until AOU_REQUIRED files are present

### PR DESCRIPTION
**Pull request by jp3477**

### CODE REVIEW

**Comment by jp3477** [comment]
> Am I missing something or this this only ensure the newest file is > 5 minutes old? I don't see any logic to ensure all files are present.

You're correct, thanks for catching that. It should be fixed now.

**Comment by ksdkalluri** [comment]
@jp3477 Are you still working on this?

**Comment by jp3477** [comment]
> @jp3477 Are you still working on this?

No, unless you think it requires more changes.

**Review by ksdkalluri — CHANGES_REQUESTED** [review]

File: data_steward/validation/main.py [hunk 1]
```diff
@@ -753,16 +753,28 @@ def list_submitted_bucket_items(folder_bucketitems):
...
             retention_time = datetime.timedelta(days=object_retention_days)
             retention_start_time = datetime.timedelta(days=1)
-            age_threshold = created_date + retention_time - retention_start_time
-            if age_threshold > today:
+            upper_age_threshold = created_date + retention_time - retention_start_time
+
+            if upper_age_threshold > today:
                 files_list.append(file_name)
+
+            if basename(file_name) in AOU_REQUIRED_FILES:
```
[the remark this answers is not in the document]
Comment (jp3477) [reply data_steward/validation/main.py:769]: Yes, you're right, not sure how I missed that. Adding that fix now.

[the remark this answers is not in the document]
Comment (ksdkalluri) [reply data_steward/validation/main.py:769]: @jp3477 my bad the condition should something be like if all files in AOU_REQUIRED_FILES are in files_list then move to next step.

**Review by ksdkalluri — DISMISSED** [review]
Good change. I do want to see if the integration tests pass before merging though.

**Review by ksdkalluri — APPROVED** [review]
All the tests that should work are working fine. Thanks for the changes!!

**Pull request merged**